File: src/main/java/com/mantenimiento/equipomedico/app/service/RepresentanteService.java
package com.mantenimiento.equipomedico.app.service;

import com.mantenimiento.equipomedico.app.entidad.Representante;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public interface RepresentanteService {

    /**
     * Creación de un nuevo representante.
     *
     * @param representante
     * @return
     */
    Representante create(Representante representante);

    /**
     * Edición de un representante existente.
     *
     * @param representante
     * @return
     */
    Representante update(Representante representante);

    /**
     * Obtiene el equipo mediante su id.
     *
     * @param id
     * @return
     */
    Representante get(Long id);

    /**
     * Obtiene todos los representantes.
     *
     * @return
     */
    List<Representante> getAll();

}

File: client-app/src/app/service/representante.service.ts
import {Injectable} from '@angular/core';
import {environment} from '../../environments/environment';
import {ApiRequestService} from './api-request.service';
import {Observable} from 'rxjs';
import {Representante} from '../domain/representante';

@Injectable({
  providedIn: 'root'
})
export class RepresentanteService {

  private urlRepresentantes = environment.service_uri + '/representantes';

  constructor(private apiRequest: ApiRequestService) {
  }


  /**
   * Se obtiene la lista de todos los representantes.
   */
  getAllRepresentantes(): Observable<Representante[]> {
    const url = this.urlRepresentantes;
    return this.apiRequest.get(url);
  }

  /**
   * Se obtiene un representante por su ID.
   * @param representanteId
   */
  getRepresentanteById(representanteId: number): Observable<Representante> {
    const url = this.urlRepresentantes + '/' + representanteId;
    return this.apiRequest.get(url);
  }

  /**
   * Se crea un nuevo representante
   * @param representante
   */
  crearRepresentanteEquipo(representante: Representante): Observable<Representante> {
    const url = this.urlRepresentantes;
    return this.apiRequest.post(url, representante);
  }

  /**
   * Se editan los datos de un representante existente.
   * @param representante
   */
  editarRepresentanteoEquipo(representante: Representante): Observable<Representante> {
    const url = this.urlRepresentantes;
    return this.apiRequest.put(url, representante);
  }

}
